Today Extension/TodayViewController.swift

```diff
@@ -0,0 +1,226 @@
+    private func setUpRoutesTableView() {
+        routesTable.delegate = self
+        routesTable.dataSource = self
+        routesTable.register(TodayExtensionCell.self, forCellReuseIdentifier: Constants.TodayExtension.contentCellIdentifier)
+        routesTable.register(TodayExtensionErrorCell.self, forCellReuseIdentifier: Constants.TodayExtension.errorCellIdentifier)
+        routesTable.register(LoadingTableViewCell.self, forCellReuseIdentifier: Constants.TodayExtension.loadingCellIdentifier)
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let isCompact = extensionContext?.widgetActiveDisplayMode == .compact
+
+        if (isCompact || favorites.isEmpty || routes.isEmpty || !didFetchRoutes) {
+            // if: in compact mode, no favorites added, failed to load routes, or in loading state
+            return 1
+        } else {
+            return favorites.count
+        }
+
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 110.0
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if (favorites.count != 0) {
+            if (routes.isEmpty) {
+                if (didFetchRoutes) { // tried to get routes, but none retrieved
+                    let cell = tableView.dequeueReusableCell(withIdentifier: Constants.TodayExtension.errorCellIdentifier, for: indexPath) as! TodayExtensionErrorCell
+                    cell.mainLabel.text = Constants.TodayExtension.unableToLoad
+                    cell.selectionStyle = .none
+                    return cell
+                } else { // still fetching routes
+                    let cell = tableView.dequeueReusableCell(withIdentifier: Constants.TodayExtension.loadingCellIdentifier, for: indexPath) as! LoadingTableViewCell
+                    cell.selectionStyle = .none
+                    return cell
+                }
+            }
+            let cell = tableView.dequeueReusableCell(withIdentifier: Constants.TodayExtension.contentCellIdentifier, for: indexPath) as! TodayExtensionCell
+            routes[indexPath.row]?.formatDirections(start: Constants.General.currentLocation, end: favorites[indexPath.row])
+            cell.setUpCell(route: routes[indexPath.row], destination: favorites[indexPath.row])
+            cell.selectionStyle = .none
+            return cell
+        }
+
+        // else: favorites = 0
+        let cell = tableView.dequeueReusableCell(withIdentifier: Constants.TodayExtension.errorCellIdentifier, for: indexPath) as! TodayExtensionErrorCell
+        cell.boldLabel.text = Constants.TodayExtension.addFavorite
+        cell.mainLabel.text = Constants.TodayExtension.showTrips
+        cell.selectionStyle = .none
+        return cell
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+
+        var stringURL: String
```

Comment:
nittiest-of-picks: you can set `stringURL` to be "ithaca-transit://" (should also maybe be in Constants.swift), and then use `+=` later on